libs/vm/include/vm/defs.hpp

```diff
@@ -903,20 +903,23 @@ struct Variant
   }
 };
 
-struct TemplateParameter : public Variant
+template <std::size_t ID = 0>
+struct TemplateParameterT : public Variant
 {
   using Variant::Variant;
+  TemplateParameterT(TemplateParameterT const &) = default;
+  TemplateParameterT(TemplateParameterT &&)      = default;
+  TemplateParameterT(Variant const &other)
+    : Variant{other} {};
```

Comment:
Unnecessary semicolon, ditto below.
```suggestion
    : Variant{other} {}
```